src/metabase/native_query_analyzer/replacement.clj

```diff
@@ -29,6 +29,13 @@
   [raw-query]
   (first-unique raw-query #(str (gensym "metabase_sentinel_table_"))))
 
+(defn- gen-snippet-sentinel
+  [raw-query {:keys [content]}]
+  (let [delimited-snippet (fn [sentinel snippet-contents]
+                            (format "/* snippet_start_%s */ %s /* snippet_end_%s */"
+                                    sentinel snippet-contents sentinel))]
```

Comment:
It feels like this could be broken out into a top level `get-snippet-sentinel*` helper, to save some nested and avoid creating lambdas. I don't think inlining it would hurt readability much either.